### PR TITLE
Update interest calculation for collateral custodies

### PR DESCRIPTION
**Pull request by askibin**

### CODE REVIEW

**Comment by mzs-dev** [comment]
Don't think this fixes the issue with misappropriated computation of interest.
The proposed change here: https://github.com/askibin/perpetuals-fork/blob/86c3f3498019807a89df9f54ab85be76008812dc/programs/perpetuals/src/state/custody.rs#L497-L498 does account for proper tracking of interest accrued on collateral custody in case of shorts or virtual asset positions.
However we're computing the wrong interest amount in case of shorts here: https://github.com/askibin/perpetuals-fork/blob/86c3f3498019807a89df9f54ab85be76008812dc/programs/perpetuals/src/state/custody.rs#L444
and updating the stats here: https://github.com/askibin/perpetuals-fork/blob/86c3f3498019807a89df9f54ab85be76008812dc/programs/perpetuals/src/state/custody.rs#L453
Which is included in pool aum calculations here: https://github.com/solana-labs/perpetuals/blob/2b715801f5531c7d75800acfe1e9f25c18d20700/programs/perpetuals/src/state/pool.rs#L756-L799

File: programs/perpetuals/src/state/custody.rs [hunk 1]
```diff
@@ -493,6 +493,10 @@ impl Custody {
 
         // update collateral custody for interest tracking
         if let Some(custody) = collateral_custody {
+            // compute accumulated interest
+            let collective_position = custody.get_collective_position(position.side)?;
```
Comment (mzs-dev) [inline programs/perpetuals/src/state/custody.rs:497]: need to add additional changes to pool aum calculations to fix the issue

**Review by mzs-dev — CHANGES_REQUESTED** [review]

**Comment by askibin** [comment]
agree, the interest was counted twice for shorts/virtual, in both custody and collateral_custody, updated

**Review by mzs-dev — APPROVED** [review]
Looks good now...
I'd still want to have a seperate market state to track

File: programs/perpetuals/src/instructions/open_position.rs [hunk 2]
```diff
@@ -250,6 +250,7 @@ pub fn open_position(ctx: Context<OpenPosition>, params: &OpenPositionParams) ->
     position.side = params.side;
     position.price = position_price;
     position.size_usd = size_usd;
+    position.borrow_size_usd = size_usd;
```
[the remark this answers is not in the document]
Comment (askibin) [reply programs/perpetuals/src/instructions/open_position.rs]: updated

**Pull request merged**